src/SubmarineHttp/SubmarineHttpClient.ts

```diff
@@ -0,0 +1,82 @@
+import HttpClient from "../Http/HttpClient";
+import * as StatusCodeMap from "./SubmarineStatusCodeMap";
+import SubmarineError from "./Errors/SubmarineError";
+import SubmarineExceptionalError from "./Errors/SubmarineExceptionalError";
+import ISubmarineExceptionResponse from "./ISubmarineExceptionResponse";
+import ISubmarineValidationResponse from "./ISubmarineValidationResponse";
+import SubmarineValidationError from "./Errors/SubmarineValidationError";
+import HttpClientMethod from "../Http/HttpClientMethod";
+
+export default class SubmarineHttpClient extends HttpClient {
+    private readonly _version: string;
+
+    constructor(version: string) {
+        super();
+
+        this._version = version;
+    }
+
+    public get<TResponse>(url: string, options?: RequestInit): Promise<TResponse> {
+        return this._performSubmarineHttpRequest<void, TResponse>(HttpClientMethod.GET, url, undefined, options);
+    }
+
+    public post<TRequest, TResponse>(url: string, body: TRequest, options?: RequestInit) {
+        return this._performSubmarineHttpRequest<TRequest, TResponse>(HttpClientMethod.POST, url, body, options)
+    }
+
+    public put<TRequest, TResponse>(url: string, body: TRequest, options?: RequestInit) {
+        return this._performSubmarineHttpRequest<TRequest, TResponse>(HttpClientMethod.PUT, url, body, options);
+    }
+
+    public delete(url: string, options?: RequestInit) {
+        return this._performSubmarineHttpRequest<void, void>(HttpClientMethod.DELETE, url, undefined, options);
+    }
+
+    protected _generateUrl(url: string) {
+        return `${this._version}/${url}`;
+    }
+
+    protected async _performSubmarineHttpRequest<TRequest, TResponse>(method: string, url: string, body?: TRequest, options: RequestInit = {}): Promise<TResponse>  {
+        const fullUrl = this._generateUrl(url);
+        options.body = JSON.stringify(body);
```

Comment:
I feel like this is more of an option for the HTTP method our API supports, rather than an expectation.